crates/swc_common/src/syntax_pos/hygiene.rs

```diff
@@ -61,15 +61,44 @@ struct MarkData {
     is_builtin: bool,
 }
 
+/// List of proxy calls injected by the host in the plugin's runtime context.
+/// When related calls being executed inside of the plugin, it'll call these
+/// proxies instead which'll call actual host fn.
+extern "C" {
+    // Instead of trying to copy-serialize `Mark`, this fn directly consume
+    // inner raw value as well as fn and let each context constrcuts struct
+    // on their side.
+    fn __mark_fresh_proxy(mark: i32) -> i32;
+}
+
 impl Mark {
+    #[allow(unused)]
     pub fn fresh(parent: Mark) -> Self {
-        HygieneData::with(|data| {
+        // Note: msvc tries to link against proxied fn for normal build,
+        // have to limit build target to wasm only to avoid it.
+        #[cfg(feature = "plugin-mode")]
+        #[cfg(target_arch = "wasm32")]
+        return Mark(unsafe {
+            __mark_fresh_proxy(
+                parent
+                    .as_u32()
+                    .try_into()
+                    .expect("Should able to convert mark into i32"),
+            )
+            .try_into()
+            .expect("Should able to convert i32 into mark")
+        });
+
+        #[cfg(not(feature = "plugin-mode"))]
```

Comment:
I think we need `all(not(feature = "plugin-mode"), not(target_arch = "wasm32"))`.

Reply:
It's required for testing while developement.

e.g. User build `swc_common` with `plugin-mode` (because the plugin code depends on `swc_plugin`) and it's native default target instead of wasm.

Reply:
(`cargo test`)

Reply:
I honestly would like to restrict users of `swc_plugin` always specify proper target for build / test and would not reach this path, but yes, that'll be somewhat destructive at this moment.